src/import.js

```diff
@@ -0,0 +1,43 @@
+import { lib } from './lib.js';
+
+/**
+ * Imports functionality from cornerstoneTools for use in external packages/plugins.
+ *
+ * @param  {string} uri the import path for the entity to import.
+ * @return {Class|Object|Function} The entity requested.
+ */
+export default function (uri) {
+  const splitUri = uri.split('/');
+  const depth = splitUri.length;
+
+  let [namespace, directory, item] = splitUri;
+
+  if (depth > 3) {
+    console.warn(`${uri} doesn't exist, too deep.`);
+  }
+
+  // Check that whatever is being requested exists.
+  if (depth >= 1 && !lib[namespace]) {
+    console.warn(`Namespace ${namespace} does not exist.`);
+    return;
+  }
+
+  if (depth >= 2 && !lib[namespace][directory]) {
+    console.warn(`${namespace}/${directory} does not exist.`);
+    return;
+  }
+
+  if (depth === 3 && !lib[namespace][directory][item]) {
+    console.warn(`${namespace}/${directory}/${item} does not exist.`);
+    return;
+  }
+
+  switch (depth) {
+    case 1:
+      return lib[namespace];
+    case 2:
+      return lib[namespace][directory];
+    case 3:
+      return lib[namespace][directory][item];
+  }
```

Comment:
I don't think we need to get this crazy. Most namespaces should be as simple as:

- Type of thing: `mixin`
- Name of the thing: `myMagicalMixin`

And ideally, it should always be passed in as a single string `mixin/myMagicalMixin`

That alone should ensure uniqueness, and is a perfectly fine key for an associative array. We can avoid all of the splitting and just return whatever exists at that location, or a warning if the item doesn't exist. Only need to make this as complex as it needs to be.

Reply:
Hmm sure, I just thought it'd be nice to be able to grab a whole plugin object. I felt it could be useful when making a tool that wants to import all of the functionality from a different plugin. I guess the boilerplate isn't too heavy if you have to import each item.

I was just thinking from a DX perspective, as a developer making a new plugin, it'd be nice to import a useful looking plugin, log the object to the console and see whats there. Just wondering if it may be a bit annoying to have to go to the actual plugin source and see what items are present there?